src/vstest.console/Processors/CollectArgumentProcessor.cs

```diff
@@ -113,16 +118,45 @@ public void Initialize(string argument)
                     argument));
             }
 
-            if(InferRunSettingsHelper.IsTestSettingsEnabled(this.runSettingsManager.ActiveRunSettings.SettingsXml))
+            if (InferRunSettingsHelper.IsTestSettingsEnabled(this.runSettingsManager.ActiveRunSettings.SettingsXml))
             {
                 throw new SettingsException(string.Format(CommandLineResources.CollectWithTestSettingErrorMessage, argument));
             }
             AddDataCollectorToRunSettings(argument, this.runSettingsManager);
         }
 
+        /// <summary>
+        /// We try to fix inproc coverlet codebase searching coverlet.collector.dll assembly inside adaptersPaths
+        /// </summary>
+        private void FixCoverletInProcessCollectorCodeBase()
```

Comment:
Dont fully understand why we need this? Is this so older versions of coverlet still work?

Reply:
It's not related to old version, we need to update codebase with full path to avoid the need to pass custom Coverlet adapter path to host, so my idea is to fix path during settings creation

Reply:
ah I see what's going on. Can we not fix the logic that generates these settings in this file?

Reply:
Mmm it's not clear what you mean. Coverlet is injected through --collect:`XPlat Code Coverage` so no xml setting file is provided, but it's generated on the fly in this class, but the nuget folder path is dynamic and it's injected through coverlet collector target file https://github.com/tonerdo/coverlet/blob/master/src/coverlet.collector/build/netstandard1.0/coverlet.collector.targets

Reply:
We need to do this on execute because we have to be sure that adapters element was loaded by adapter argument processor, so we can search coverlet lib inside adapters path directories.

Reply:
I see. That makes sense. 
There's probably another way of fixing this. All the processors have a priority associated which dictates which of them need to execute earlier. In this case the CollectArgumentProecessor should have a higher priority compared to the TestAdapterPathArgumentProcessor so you could just naturally make the change in the initialize without needing to fix it up later. 
So [this](https://github.com/MarcoRossignoli/vstest/blob/passcoverletcodebase/src/vstest.console/Processors/TestAdapterPathArgumentProcessor.cs#L89) should ideally be `ArgumentProcessorPriority.TestAdapterPath`.
@singhsarab - does that make sense?

Reply:
Ideally I'd like for this to be more general so any nuget based extension can easily setup configuration with the test platform. I've logged #2294 to track that down.

Reply:
> So this should ideally be ArgumentProcessorPriority.TestAdapterPath.


I see, btw this is a bit weak, like the other way should be documented and a future update could silently break loading(maybe we should add a smoke test for coverlet).
Another idea could be share(expose) the portion of code that parse adapter arg and use in both place in init.

Reply:
Sure, arguments as a read only list would work as well. We'd need to be vary about re-parsing in that case.
I absolutely agree that we need a smoke test for coverlet so this continues to work. Looks like those priorities were added for scenarios like so but just haven't been used for some reason.

Reply:
If @singhsarab agree I could opt for priority way, waiting(if necessary or give the green light).
I'll open a new issue for smoke test.

Reply:
@AbhitejJohn @singhsarab do we go for priority+init?

Reply:
Yeah, I'd vote for that format since it already exists and is used across. We'd want to add a test to ensure that this order is maintained.

Reply:
Moved to init and updated proprity for `TestAdapterPathArgumentProcessor`